src/rbd.cc

```diff
@@ -2470,6 +2469,34 @@ static int do_show_status(librados::IoCtx &io_ctx, librbd::Image &image,
   return 0;
 }
 
+static int do_rename(librbd::RBD &rbd, librados::IoCtx& io_ctx,
+		     librbd::Image &image, const char *imgname, const char *destname)
+{
+  int r = rbd.open_read_only(io_ctx, image, imgname, NULL);
+  if (r < 0) {
+    cerr << "rbd: error opening image " << imgname << ": "
+         << cpp_strerror(-r) << std::endl;
+    return -r;
+  }
+  bool watch_exists = false;
+  r = do_show_status(io_ctx, image, imgname, NULL, &watch_exists);
```

Comment:
@xinze, probably what @jdurgin means is

1. to extract the list_watchers related part into a separate helper function. 
2. let `do_show_status()` use it instead
3. let `do_rename()` use it